server/src/com/thoughtworks/go/config/GoConfigDao.java

```
@@ -88,13 +88,20 @@ public String md5OfConfigFile() {
     }
 
     public void updateConfig(EntityConfigUpdateCommand command, Username currentUser) {
-        LOGGER.info("Config update for pipeline request by {} is in queue - {}", currentUser, command);
+        LOGGER.info("Config update for entity request by {} is in queue - {}", currentUser, command);
         synchronized (GoConfigWriteLock.class) {
-            LOGGER.info("Config update for pipeline request by {} is being processed", currentUser);
-            if (!command.canContinue(cachedConfigService.currentConfig())) {
-                throw new ConfigUpdateCheckFailedException();
+            try {
+                LOGGER.info("Config update for entity request by {} is being processed", currentUser);
+                if (!command.canContinue(cachedConfigService.currentConfig())) {
+                    throw new ConfigUpdateCheckFailedException();
+                }
+                cachedConfigService.writeEntityWithLock(command, currentUser);
+            } catch (Exception e) {
+                LOGGER.info("Config update for entity failed", e);
```

need to throw the exception